lib/portlayer/storage/vsphere/datastore.go

```diff
@@ -301,3 +303,26 @@ func (d *Datastore) mkRootDir(ctx context.Context, rootdir string) error {
 func (d *Datastore) rootDir() string {
 	return strings.SplitN(d.RootURL, " ", 2)[1]
 }
+
+// Parse the datastore format ([datastore1] /path/to/thing) to groups.
+var datastoreFormat = regexp.MustCompile(`^\[([\w\d\(\)\s]+)\]`)
+var pathFormat = regexp.MustCompile(`\s[\/\w]+$`)
+
+// Converts `[datastore] /path` to URL
+func DatastoreToURL(ds string) (*url.URL, error) {
+	u := new(url.URL)
+	u.Host = strings.TrimSuffix(strings.TrimPrefix(datastoreFormat.FindString(ds), "["), "]")
```

Comment:
The `[]`s are part of the pattern because they indicate, of course, where the datastore name is. As far as I could tell (maybe I'm wrong), all of the Go regex methods include them as part of the match because they're part of the pattern. So I trim them off.

Same with the `\s` in `pathFormat`. Needs to be there, because that's what makes the rest of the pattern unique from `datastoreFormat`, but then must be trimmed.